src/components/TextInput/TextInput.tsx

```diff
@@ -1,106 +1,101 @@
+  id?: string;
+  /** Populates the HelperErrorText for the error state */
+  invalidText?: string;
+  /** Adds the `disabled` and `aria-disabled` prop to the input when true */
+  isDisabled?: boolean;
+  /** Adds errored styling to the input/textarea and helper text elements */
+  isInvalid?: boolean;
+  /** Will add `required` and `aria-required` props to the input/textarea elements */
+  isRequired?: boolean;
   /** Provides text for a `Label` component if `showLabel` is set to true;
-   * populates a `aria-label` sttribute if `showLabel` is set to false. */
+   * populates an `aria-label` attribute if `showLabel` is set to false. */
   labelText: string;
+  /** The action to perform on the `input`/`textarea`'s onChange function  */
+  onChange?: (
+    event:
+      | React.ChangeEvent<HTMLInputElement>
+      | React.ChangeEvent<HTMLTextAreaElement>
+  ) => void;
+  /** Populates the placeholder for the input/textarea elements */
+  placeholder?: string;
   /** Offers the ability to show the label onscreen or hide it. Refer to the
    * `labelText` property for more information. */
   showLabel?: boolean;
   /** Offers the ability to show the "Required"/"Optional" label onscreen or
    * hide it. True by default. */
   showOptReqLabel?: boolean;
-  /** Populates the HelperErrorText for standard state */
-  helperText?: string;
-  /** Populates the HelperErrorText for error state */
-  invalidText?: string;
-  /** Populates the placeholder of the input */
-  placeholder?: string;
-  /** Populates the value of the input */
+  /** HTML Input types as defined by MDN: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input */
+  type?: TextInputTypes;
+  /** Populates the value of the input/textarea elements */
   value?: string;
-  /** className you can add in addition to 'textinput' */
-  className?: string;
-  /** ID that other components can cross reference for accessibility purposes */
-  id?: string;
-  /** Helper for modifiers array; adds 'errored' styling */
-  errored?: boolean;
-  /** Adds the 'disabled' prop to the input when true */
-  disabled?: boolean;
-  /** Modifiers array for use with BEM. See how to work with modifiers and BEM
-   * here: http://getbem.com/introduction/ */
-  modifiers?: string[];
-  /** The action to perform on the `<input>`'s onChange function  */
-  onChange?: (
-    event:
-      | React.ChangeEvent<HTMLInputElement>
-      | React.ChangeEvent<HTMLTextAreaElement>
-  ) => void;
 }
 
-/** The type used for `ref`s. We want to extend both `input` and `textarea`
+/**
+ * The type used for `ref`s. We want to extend both `input` and `textarea`
  * since both are available to create through `TextInput`.
  */
 export type TextInputRefType = HTMLInputElement & HTMLTextAreaElement;
 
-/** Renders either an `input` element with a specified type or a `textarea`
- * element. All types will render an accessible Label component and an optional
- * HelperErrorText component.
+/**
```

Comment:
I've noticed this elsewhere as well.  Comments using this notation are showing on the story `Docs` screen.  While the text is not inaccurate, it is redundant and should be hiden from the `Docs` screen.

Reply:
Yeah, I did this on purpose because even though it's dev comment documentation, maybe devs don't check the file and just check Storybook. So it's probably best to replicate and it's easy to with the `Description` component in Storybook.

Reply:
I would like to consider Storybook as the source of truth.  But do you think there is a chance that devs don't check Storybook?